src/object.rs

```diff
@@ -1490,6 +1496,52 @@ impl<T: ObjectType> ObjectExt for T {
         }
     }
 
+    fn set_qdata(&self, key: Quark, value: Box<dyn Any>) {
+        extern "C" fn drop_value(ptr: glib_sys::gpointer) {
+            if !ptr.is_null() {
+                let value: Box<dyn Any> = unsafe { Box::from_raw(ptr as *mut dyn Any) };
+                std::mem::drop(value)
+            }
+        }
+
+        let ptr = Box::into_raw(Box::new(value)) as glib_sys::gpointer;
+        unsafe {
+            gobject_sys::g_object_set_qdata_full(
+                self.as_object_ref().to_glib_none().0,
+                key.to_glib(),
+                ptr,
+                Some(drop_value),
+            );
+        }
+    }
+
+    fn get_qdata(&self, key: Quark) -> Option<&dyn Any> {
+        unsafe {
+            let ptr = gobject_sys::g_object_get_qdata(
+                self.as_object_ref().to_glib_none().0,
+                key.to_glib(),
+            );
+            (ptr as *const Box<dyn Any>).as_ref().map(|b| &**b)
```

Comment:
Where does the `Option` come from here? An explicit `NULL` check would probably be good for code readability.

Reply:
Done